cmd/sonobuoy/app/run.go

```diff
@@ -36,7 +37,11 @@ type runFlags struct {
 	genFile       string
 }
 
-var runflags runFlags
+var (
+	runflags runFlags
+
+	allowedGenFlagsWithRunFile = []string{"kubeconfig", "context"}
```

Comment:
This was the big ? for me. Is this the full list of things that users should be able to specify while also using `--file`?

I couldn't find others that weren't meant to customize the run.

Reply:
Also, its unfortunate but we do just have to hardcode the names unless we overengineer an alternative solution.

We do not typically change the flag names so I dont expect it will be a problem though.

Reply:
I think these are the only flags that need to be excluded. Yeah, I guess you could have the flag names as constants so that they can be included here and reference the same strings. The other alternative I could think of was extracting these config values so that they are added explicitly to the `run` command rather than being included through the `GenFlagSet` so that you then error on _any_ changed flag within that set. But it's probably an overengineered solution :)

Reply:
I went with just making a constant; it gets a bit tricky to pull out the values from the genFlagSet since it also stores those flags within the data structure; I just didn't want to have anything cascade. Not against a change in the future if this annoys us though.